server.go

```diff
@@ -27,6 +27,17 @@ type comment struct {
 	value string
 }
 
+type NotificationType int
+
+const (
+	Connect NotificationType = iota
+	Disconnect
+)
+
+type Listener interface {
+	Notify(channel string, notificationType NotificationType, args ...interface{})
```

Comment:
Would making `Notification` and interface with a `NotificationType()` method work for this?  I think that might be more go-ish.